File: GFG POTD/011-October-2024/021-split-the-array.cpp
// https://www.geeksforgeeks.org/problems/split-the-array0238/1




//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;
// } Driver Code Ends

// User function template for C++
class Solution {
  public:
    int countgroup(vector<int>& arr) {
        int n = arr.size();
        int xorVal = 0;
        for (auto& num : arr) {
            xorVal ^= num;
        }
        
        if (xorVal != 0) return 0;
        
        return (1 << (n - 1)) - 1;
    }
};

//{ Driver Code Starts.
int main() {
    string ts;
    getline(cin, ts);
    int t = stoi(ts);
    while (t--) {
        vector<int> arr;
        string input;
        getline(cin, input);
        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }
        Solution obj;
        int ans = obj.countgroup(arr);
        cout << ans << endl;
    }
}
// } Driver Code Ends

File: GFG POTD/011-October-2024/021-split-the-array.java
// https://www.geeksforgeeks.org/problems/split-the-array0238/1




//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.lang.*;
import java.util.*;
// } Driver Code Ends
// User function Template for Java

class Solution {
    public static int countgroup(int arr[]) {
        int n = arr.length;
        int xorV = 0;
        for (int num : arr) {
            xorV ^= num;
        }
        
        if (xorV != 0) return 0;
        
        return (1 << (n - 1)) - 1;
    }
}


//{ Driver Code Starts.
class GFG {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        while (t-- > 0) {
            // int k = Integer.parseInt(br.readLine());
            String line = br.readLine();
            String[] tokens = line.split(" ");

            // Create an ArrayList to store the integers
            ArrayList<Integer> array = new ArrayList<>();

            // Parse the tokens into integers and add to the array
            for (String token : tokens) {
                array.add(Integer.parseInt(token));
            }

            int[] arr = new int[array.size()];
            int idx = 0;
            for (int i : array) arr[idx++] = i;
            Solution obj = new Solution();
            int ans = obj.countgroup(arr);
            System.out.println(ans);
        }
    }
}
// } Driver Code Ends